redfish-core/lib/oem/ibm/pcie_topology_refresh.hpp

```diff
@@ -55,18 +136,28 @@ inline void
  * @return None.
  */
 inline void
-    setPCIeTopologyRefresh(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+    setPCIeTopologyRefresh(const crow::Request& req,
+                           const std::shared_ptr<bmcweb::AsyncResp>& aResp,
                            const bool state)
 {
     BMCWEB_LOG_DEBUG << "Set PCIe Topology Refresh status.";
     crow::connections::systemBus->async_method_call(
-        [aResp](const boost::system::error_code ec) {
+        [&req, aResp](const boost::system::error_code ec) {
             if (ec)
             {
                 BMCWEB_LOG_DEBUG << "PCIe Topology Refresh failed." << ec;
                 messages::internalError(aResp->res);
                 return;
             }
+            count = 0;
+            pcieTopologyRefreshTimer =
+                std::make_unique<boost::asio::steady_timer>(*req.ioService);
+            pcieTopologyRefreshTimer->expires_after(std::chrono::seconds(1));
+            pcieTopologyRefreshTimer->async_wait(
+                [timer = pcieTopologyRefreshTimer.get(),
+                 aResp](const boost::system::error_code ec) {
+                    pcieTopologyRefreshWatchdog(ec, timer, aResp, &count);
```

Comment:
A0: yes. Because when the user sets that value true from there, it hast to start 8 second counter.
A1: if we get a second request then the first request gone get deleted. because it is unique_ptr.
A2: yes, second call set setPCIETopologyRefresh that is possible. but then the first user request has gone get deleted.

so this code is perfectly based on requirements, because if I am not wrong only bmcweb UI will make this refresh request. and then want to hold that request until refresh not happened.
for multiple users wants to use this feature then this code is not perfect. and I think we do not need to make this work for multiple users @gtmills please conform to that. 
if we like to make this available for multiple users then either we need more robust code or we have to move for the task(which do two replies).